src/ProductData.php

```diff
@@ -248,6 +248,21 @@ public function equals($otherProductData): bool
         $this->tierPrices->equals($otherProductData->tierPrices) &&
         $this->productLinks->equals($otherProductData->productLinks) &&
         ($this->storeCode === $otherProductData->storeCode);
+
+        $mediaEquals = true;
+        if (
+            (isset($this->mediaGalleryEntries) && isset($otherProductData->mediaGalleryEntries)) &&
+            (null == $this->mediaGalleryEntries && null !== $otherProductData->mediaGalleryEntries) ||
```

Comment:
is this supposed to be == or === ?? considering that ```(null == [])``` is true

Reply:
^ @thisisandrew

Reply:
@thisisandrew Good stuff. But I was wondering if you could look at the coverage in codecov the ProductData Class lost some coverage (-.183%) and its around the section of your implementation. I reckon updating the tests and refactoring that logic to a private method is all thats needed really.

Reply:
https://codecov.io/gh/snowio/magento2-data-model/compare/a5e1d2ea49f079f11d800e7d517c786d04622bd8...9c3b6952653ac954db61983c70da5e2675c82184/src/src/ProductData.php?before=src/ProductData.php#L255